### DISCUSSION TITLE
Verbal socket error codes

### DISCUSSION THREAD

**Discussion by YuriiMaiboroda** [make-all/tuya-local#474] [Ideas]
I found a [description](https://developer.tuya.com/en/docs/iot/product-standard-function-introduction?id=K9tp15ceh63gr#title-4-Energy%20metering) of what error codes are.

|Bitmap | Fault value |
|-- | --|
|Bit0 | ov_cr |
|Bit1 | ov_vol |
|Bit2 | ov_pwr |
|Bit3 | ls_cr |
|Bit4 | ls_vol |
|Bit5 | ls_pow |

This can be used to show the user the reason for the failure.
Personally, I have not yet had failures due to overload. But I think the inscription in words that there was too much current or too little voltage would be better than a mark that there is an error and just it's number.

**Comment by make-all** [comment]
Yes, if we can figure out the meanings of the bitfields, it is better to have descriptions, and several devices have this already.
